src/librustc_trans/monomorphize.rs

```diff
@@ -143,6 +143,12 @@ fn resolve_associated_item<'a, 'tcx>(
                 substs: rcvr_substs
             }
         }
+        traits::VtableBuiltin(ref data) => {
+            Instance {
+                def: ty::InstanceDef::BuiltinShim(def_id, data.ty),
```

Comment:
You don't need to store `ty` anywhere - it is available as `trait_ref.self_ty()`.